File: front-end/src/app/service-models/app/v1/client/client-generic-payroll-component-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { ComponentCalculationTypeSM } from '../../enums/component-calculation-type-s-m.enum';
import { ComponentPeriodTypeSM } from '../../enums/component-period-type-s-m.enum';

export class ClientGenericPayrollComponentSM extends CoinManagementServiceModelBase<number> {
    name!: string;
    description!: string;
    percentage!: number;
    componentCalculationType!: ComponentCalculationTypeSM;
    componentPeriodType!: ComponentPeriodTypeSM;
    clientCompanyDetailId!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientGenericPayrollComponentSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientGenericPayrollComponentSM : SiffrumPayrollServiceModelBase<int>
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public float Percentage { get; set; }

        public ComponentCalculationTypeSM ComponentCalculationType { get; set; }

        public ComponentPeriodTypeSM ComponentPeriodType { get; set; }

        public int ClientCompanyDetailId { get; set; }
    }
}
